LgiDisplaySettings/LgiDisplaySettings.cpp

```diff
@@ -0,0 +1,359 @@
+/**
+* If not stated otherwise in this file or this component's LICENSE
+* file the following copyright and licenses apply:
+*
+* Copyright 2021 RDK Management
+* Copyright 2021 Liberty Global Service B.V.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+#include "LgiDisplaySettings.h"
+#include <algorithm>
+#include "host.hpp"
+#include "exception.hpp"
+#include "videoOutputPort.hpp"
+#include "videoOutputPortType.hpp"
+#include "videoOutputPortConfig.hpp"
+#include "videoResolution.hpp"
+#include "audioOutputPort.hpp"
+#include "audioOutputPortType.hpp"
+#include "audioOutputPortConfig.hpp"
+#include "dsError.h"
+
+using namespace std;
+
+namespace WPEFramework {
+
+    namespace Plugin {
+
+        SERVICE_REGISTRATION(LgiDisplaySettings, 1, 0);
+
+	extern void setResponseArray(JsonObject& response, const char* key, const vector<string>& items);
+
+        static bool parseQBool(const std::string& str)
+        {
+            // https://doc.qt.io/qt-5/qvariant.html#toBool:
+            // 'Returns true if (...) lower-case content is not one of the following: empty, "0" or "false"; otherwise returns false.'
+            string lowercase_string{};
+            std::transform(str.begin(),
+                           str.end(),
+                           std::back_inserter(lowercase_string),
+                           ::tolower);
+            return !(lowercase_string.empty() || lowercase_string == "false" || lowercase_string == "0");
+        }
+
+        LgiDisplaySettings::LgiDisplaySettings() : DisplaySettings()
+        {
+            Register("setOutputFrameRatePreference", &LgiDisplaySettings::setOutputFrameRatePreference, this);
```

Comment:
Utils::Synchro::RegisterLockedApi should probably be used for method registration (see dt/helpers/UtilsSynchroIarm.hpp) - here and further on